src/Symfony/Component/Translation/Extractor/Visitor/TransMethodVisitor.php

```diff
@@ -44,7 +44,7 @@ public function leaveNode(Node $node): ?Node
         if ('trans' === $name || 't' === $name) {
             $firstNamedArgumentIndex = $this->nodeFirstNamedArgumentIndex($node);
 
-            if (!$messages = $this->getStringArguments($node, 0 < $firstNamedArgumentIndex ? 0 : 'message')) {
+            if (!$messages = $this->getStringArguments($node, 0 < $firstNamedArgumentIndex ? 0 : 'id')) {
```

Comment:
Both the twig callables `t` and `trans` refer to the translator `trans()` method / `t` function, and all of them have '`$message`' as first attribute 🤷‍♂️ 

So i'm not sure to understand here (especially as you changed the MethodVisitor code _and_ the test at the same time)

https://github.com/symfony/twig-bridge/blob/1dad48d7e1c120aada838d6d2ae57a4c5a87390f/Extension/TranslationExtension.php#L96

https://github.com/symfony/symfony/blob/8418ca2cef59ce88c5152fc60c53517d9ccd3f3c/src/Symfony/Component/Translation/Resources/functions.php#L18

Reply:
You can try to inject TranslatorInterface to your service, call ->trans()

->trans('foo', domain: 'domain') this is extractable 

->trans(id: 'foo', domain: 'domain') this is NOT extractable